src/java/org/apache/cassandra/db/virtual/CachesTable.java

```diff
@@ -67,6 +68,19 @@ private void addRow(SimpleDataSet result, String name, CacheMetrics metrics)
               .column(RECENT_HIT_RATE_PER_SECOND, (long) metrics.hits.getFifteenMinuteRate());
     }
 
+    private void addRow(SimpleDataSet result, String name, ChunkCacheMetrics metrics)
+    {
+        result.row(name)
+              .column(CAPACITY_BYTES, metrics.capacity())
+              .column(SIZE_BYTES, metrics.size())
+              .column(ENTRY_COUNT, metrics.entries())
+              .column(REQUEST_COUNT, metrics.requests())
+              .column(HIT_COUNT, metrics.hits())
+              .column(HIT_RATIO, metrics.hitRate())
+              .column(RECENT_REQUEST_RATE_PER_SECOND, metrics.requestsFifteenMinuteRate())
+              .column(RECENT_HIT_RATE_PER_SECOND, metrics.hitsFifteenMinuteRate());
+    }
+
```

Comment:
Wanted to point out that this differs from the original; there the `CacheMetrics` constructor is explicitly calling `CachesSystemView.addRowData` to register cache metrics into the system view.

In this code ChunkCache instance metrics are added by `CachesTable.data()` using `ChunkCache.instance.metrics`, which is `ChunkCacheMetrics` interface; so by having `addRow` taking that type we will get metrics for any implementing class.

I added `ChunkCacheMetrics.requestsFifteenMinuteRate` and `hitsFifteenMinuteRate` for that; only `CodahaleChunkCacheMetrics` provides those values. `MicrometerChunkCacheMetrics` will return zeros.